internal/controller/event_filter.go

```diff
@@ -0,0 +1,91 @@
+package controller
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"golang.org/x/net/context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type Status struct {
+	LastRetryTs int64
+	RetryCount  int
+}
+
+const minInterval = 50 * time.Millisecond
+const maxInterval = 30 * time.Minute
+
+func ignoreIfBeforeRequeueDelay(
+	objToStatus func(client.Object) (Status, error),
+	requeueDelay time.Duration) predicate.Predicate {
+
+	check := func(obj client.Object) bool {
+		logger := log.FromContext(context.TODO())
+		fullName := fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName())
+		logger.Info(fmt.Sprintf("Checking %s for reconciliation", fullName))
+
+		status, err := objToStatus(obj)
+		if err != nil {
+			logger.Info(fmt.Sprintf("Failed to extract status for resource %s. Trying to reconcile", fullName))
+			return true
+		}
+		nextTickInThePast := isNextTickInThePast(status, requeueDelay)
+		if nextTickInThePast {
+			logger.Info(fmt.Sprintf("Attempting reconciliation of %s for the %d-th time",
+				fullName, status.RetryCount))
+		} else {
+			logger.Info(fmt.Sprintf("Skipping reconciliation of %s for now", fullName))
```

Comment:
shouldnt this log be a debug one yo minimalize amount of logs?

Reply:
It's not yet clear for me what's "debug log" in go-logr, but based on recommendations I've used level 4 for now